### PR TITLE
Support for scte35-2020 Pass-1

### PR DESCRIPTION
**Pull request by nehashanbhag**
Added/Updated closing logic for 0x10, 0x11, 0x22, 0x23, 0x24, 0x25, 0x26, 0x27, 0x34, 0x35, 0x3C, 0x3D, 0x42, 0x43, 0x44, 0x45

### CODE REVIEW

File: scte35/doc.go [hunk 1]
```diff
@@ -79,65 +79,85 @@ var DeviceRestrictionsNames = map[DeviceRestrictions]string{
...
+	SegDescClosingCreditStart                           = 0x26
+	SegDescClosingCreditEnd                             = 0x27
+	SegDescProviderAdvertisementStart                   = 0x30
+	SegDescProviderAdvertisementEnd                     = 0x31
+	SegDescDistributorAdvertisementStart                = 0x32
+	SegDescDistributorAdvertisementEnd                  = 0x33
+	SegDescProviderPOStart                              = 0x34
+	SegDescProviderPOEnd                                = 0x35
+	SegDescDistributorPOStart                           = 0x36
+	SegDescDistributorPOEnd                             = 0x37
```
Comment (nehashanbhag) [inline scte35/doc.go:109]: Added new signals here 0x3C - 0x40, 0x42-0x45

File: scte35/doc.go [hunk 2]
```diff
@@ -79,65 +79,85 @@ var DeviceRestrictionsNames = map[DeviceRestrictions]string{
...
+	SegDescProgramRunoverPlanned                        = 0x15
+	SegDescProgramRunoverUnplanned                      = 0x16
+	SegDescProgramOverlapStart                          = 0x17
+	SegDescProgramBlackoutOverride                      = 0x18
+	SegDescProgramStartInProgress                       = 0x19
+	SegDescChapterStart                                 = 0x20
+	SegDescChapterEnd                                   = 0x21
+	SegDescBreakStart                                   = 0x22
+	SegDescBreakEnd                                     = 0x23
+	SegDescOpeningCreditStart                           = 0x24
```
Comment (nehashanbhag) [inline scte35/doc.go:98]: Added new descriptors 0x24-0x27

File: scte35/state_test.go [hunk 3]
```diff
@@ -743,34 +706,14 @@ func TestOutInInOutIn36_37_35_37_37(t *testing.T) {
...
-	// ErrSCTE35MissingOut expected as 0x35 closed 0x36
-	if err != gots.ErrSCTE35MissingOut {
-		t.Error("ProcessDescriptor of out returned unexpected err:", err)
-	}
-	if len(closed) != 0 {
-		t.Errorf("No events should have been closed (%d were)", len(closed))
-	}
-	if len(state.Open()) != 0 {
+	if len(state.Open()) != 1 {
 		t.Errorf("There should be 0 open signal (%d)", len(state.Open()))
```
Comment (laba2346) [inline scte35/state_test.go:717]: Should this error message be updated to reflect the new conditional?

File: scte35/state_test.go [hunk 4]
```diff
@@ -510,48 +480,41 @@ func TestOutInInOutIn(t *testing.T) {
...
 		t.Errorf("ProcessDescriptor returned an error: %s", err.Error())
 	}
-	if len(closed) != 1 {
-		t.Errorf("One event should have been closed (%d were)", len(closed))
+	if len(closed) != 0 {
+		t.Errorf("No events should have been closed (%d were)", len(closed))
 	}
-	if len(state.Open()) != 1 {
+	if len(state.Open()) != 2 {
 		t.Errorf("There should be one open signals (%d)", len(state.Open()))
```
Comment (laba2346) [inline scte35/state_test.go:498]: Should this error message be updated to reflect the new conditional?

File: scte35/state_test.go [hunk 5]
```diff
@@ -303,48 +286,34 @@ func TestOutOutIn(t *testing.T) {
...
 	if err != nil {
-		t.Errorf("ProcessDescriptor returned an error: %s", err.Error())
-	}
-	if len(closed) != 0 {
-		t.Errorf("No events should have been closed (%d were)", len(closed))
+		t.Error("ProcessDescriptor of out returned unexpected err:", err)
 	}
 	if len(state.Open()) != 1 {
-		t.Errorf("There should be one open signal (%d)", len(state.Open()))
+		t.Error("Unexpected number of open signals after padOpenSignal processed.")
```
Comment (laba2346) [inline scte35/state_test.go:305]: Can we add back the `len(state.Open())`? I think it's useful information for an error message.

File: scte35/state_test.go [hunk 6]
```diff
@@ -253,48 +234,50 @@ func TestDuplicateOut(t *testing.T) {
...
-	// 0x34 - 2 - (2/2)
-	open22, e := NewSCTE35(poOpen22)
-	if e != nil {
-		t.Error("NewSCTE35(poOpen22) return err:", e)
+	// 0x34
+	ppoStartSignalBytes, _ := base64.StdEncoding.DecodeString("/DA0AAG3wyKO///wBQb/yRLo+gAeAhxDVUVJQAAAZH/PAAEOFNwICAAAAAArJkYdNAAAMcFkEw==")
+	ppoStartSignal, err := NewSCTE35(append([]byte{0x0}, ppoStartSignalBytes...))
+	if err != nil {
+		t.Error("NewSCTE35(ppoStartSignal) return err:", err)
 		t.FailNow()
```
Comment (laba2346) [inline scte35/state_test.go:257]: Thoughts on replacing these two lines with `Fatal`?
`t.Fatal("NewSCTE35(ppoStartSignal) return err:", err)`

File: scte35/state_test.go (hunk 4 above)
Comment (nehashanbhag) [reply scte35/state_test.go:498]: oops geez, bad miss. sorry! will fix!

File: scte35/state_test.go (hunk 6 above)
Comment (nehashanbhag) [reply scte35/state_test.go:257]: good point. I was doing copy pasta of old tests to write this new one. I like t.Fatal as well.

**Review by laba2346 — APPROVED** [review]

**Pull request merged**